### PR TITLE
OCPBUGS-1265: test/extended/prometheus: Consider telemeterClient.enabled

### PR DESCRIPTION
**Pull request by wking**
I'd disabled Telemetry for the bulk of the CI fleet in openshift/release@3c1da8eb20 (openshift/release#32153).  But that lead to many failures for:

    Prometheus when installed on the cluster should report telemetry if a cloud.openshift.com token is present

This pull request extends the checks for Telemetry enablement to include the monitoring-specific ConfigMap, as well as the previously-checked pull-secret token.

I'm copy/pasting a subset of the monitoring configuration structure instead of vendoring the config, because we aren't vendoring the cluster-monitoring-operator in origin today, bumping to keep such vendoring up to date would be tedious, and the monitoring config API is unlikely to shift this knob around within the structure.  It would be nice if the monitoring config schema moved into opensihft/api or somewhere else where it would be more clear that it was a cluster-admin-facing API, with the usual stability commitments, but it's not there today.

### CODE REVIEW

**Review by dgoodwin — COMMENTED** [review]
lgtm otherwise.

File: test/extended/prometheus/prometheus.go [hunk 1]
```diff
@@ -920,9 +946,38 @@ func hasPullSecret(client clientset.Interface, name string) bool {
...
+	}
+
+	return nil, nil
+}
+
+func isTelemeterClientEnabled(ctx context.Context, client clientset.Interface) (enabled error, err error) {
+	config, err := client.CoreV1().ConfigMaps("openshift-monitoring").Get(ctx, "cluster-monitoring-config", metav1.GetOptions{})
+	if err != nil {
+		if kapierrs.IsNotFound(err) {
+			return true, nil // Telemetry is enabled by default
```
[the remark this answers is not in the document]
Comment (wking) [reply test/extended/prometheus/prometheus.go]: Oops; thanks.  Fixed with fa8c329f24 -> 9ceeff2ab8.

File: test/extended/prometheus/prometheus.go [hunk 2]
```diff
@@ -920,9 +946,38 @@ func hasPullSecret(client clientset.Interface, name string) bool {
...
+		}
+		return nil, fmt.Errorf("could not retrieve monitoring configuration: %w", err)
+	}
+	if yamlConfig, ok := config.Data["config.yaml"]; !ok {
+		return nil, fmt.Errorf("openshift-monitoring/cluster-monitoring-config data lacks a config.yaml key: %v", config.Data)
+	}
+	var structuredConfig ClusterMonitoringConfiguration
+	if err := yaml.Unmarshal([]byte(invalidAPIServer), &structuredConfig); err != nil {
+		return nil, fmt.Errorf("openshift-monitoring/cluster-monitoring-config config.yaml: %w", err)
```
[the remark this answers is not in the document]
Comment (wking) [reply test/extended/prometheus/prometheus.go]: Done with fa8c329f24 -> 9ceeff2ab8.

**Comment by wking** [comment]
[Success][1]:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27422/pull-ci-openshift-origin-master-e2e-aws/1570659685793533952/artifacts/e2e-aws/openshift-e2e-test/build-log.txt | grep -B3 'skipped.*Prometheus when installed on the cluster should report telemetry'
skip [github.com/openshift/origin/test/extended/prometheus/prometheus.go:461]: Telemetry is disabled: openshift-monitoring/cluster-monitoring-config telemeterClient enabled is: false
Ginkgo exit error 3: exit with code 3

skipped: (7.3s) 2022-09-16T07:57:12 "[sig-instrumentation] Prometheus when installed on the cluster should report telemetry [Late] [Skipped:Disconnected] [Suite:openshift/conformance/parallel]"
```

and the run as a whole passed.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27422/pull-ci-openshift-origin-master-e2e-aws/1570659685793533952

**Comment by stbenjam** [comment]
This is renaming a test but the annotations have been correctly updated, and I don't see any references to it in the release repo.

/lgtm

**Pull request merged**